src/component/ResponsiveContainer.tsx

```diff
@@ -240,3 +223,62 @@ export const ResponsiveContainer = forwardRef<HTMLDivElement, Props>(
     );
   },
 );
+
+/**
+ * The `ResponsiveContainer` component is a container that adjusts its width and height based on the size of its parent element.
+ * It is used to create responsive charts that adapt to different screen sizes.
+ *
+ * This component uses the `ResizeObserver` API to monitor changes to the size of its parent element.
+ * If you need to support older browsers that do not support this API, you may need to include a polyfill.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver
+ */
+export const ResponsiveContainer = forwardRef<HTMLDivElement, Props>((props, ref) => {
+  const responsiveContainerContext = useResponsiveContainerContext();
+  if (isPositiveNumber(responsiveContainerContext.width) && isPositiveNumber(responsiveContainerContext.height)) {
+    /*
+     * If we detect that we are already inside another ResponsiveContainer,
+     * we do not attempt to add another layer of responsiveness.
+     */
+    return props.children;
+  }
+
+  const { width, height } = getDefaultWidthAndHeight({
+    width: props.width,
+    height: props.height,
+    aspect: props.aspect,
+  });
+
+  /*
+   * Let's try to get the calculated dimensions without having the div container set up.
+   * Sometimes this does produce fixed, positive dimensions. If so, we can skip rendering the div and monitoring its size.
+   */
+  const { calculatedWidth, calculatedHeight } = calculateChartDimensions(undefined, undefined, {
+    width,
+    height,
+    aspect: props.aspect,
+    maxHeight: props.maxHeight,
+  });
+
+  if (isNumber(calculatedWidth) && isNumber(calculatedHeight)) {
+    /*
+     * If it just so happens that the combination of width, height, and aspect ratio
+     * results in fixed dimensions, then we don't need to monitor the container's size.
+     * We can just provide these fixed dimensions to the context.
+     *
+     * Note that here we are not checking for positive numbers;
+     * if the user provides a zero or negative width/height, we will just pass that along
+     * as whatever size we detect won't be helping anyway.
+     */
+    return (
+      <ResponsiveContainerContextProvider width={calculatedWidth} height={calculatedHeight}>
```

Comment:
I did add this shortcut @ckifer - in case we are able to calculate specific size from the given arguments then the helper div doesn't render at all.

I haven't tested SSR though, and I am not sure what is the problem in SSR in the first place, so can't tell if this makes it better or worse.